File: Programmers/Lv3/Lv3_최고의집합.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> solution(int n, int s) {
	vector<int> answer;
	if (s / n == 0)
		return { -1 };
	else {
		int q = s / n;
		int r = s % n;
		int j = n - 1;
		for(int i=0; i<n; i++)
			answer.push_back(q);
		while (r != 0) {
			answer[j--]++;
			r--;
		}
	}
	return answer;
}

File: Programmers/Lv3/Lv3_최고의집합.py
def solution(n, s):
    answer = []
    if n>s:
        answer.append(-1)
    elif n==s:
        for i in range(0,n):
            answer.append(1)
    else:
        q = s//n
        r = s%n
        j = n-1
        for i in range(n):
            answer.append(q)
        while r!=0:
            answer[j] += 1
            j -= 1
            r -= 1
    return answer